plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeTableChangesWithAccessTracking.java

```diff
@@ -0,0 +1,96 @@
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestDeltaLakeTableChangesWithAccessTracking
+        extends AbstractTestQueryFramework
+{
+    protected static final String SCHEMA = "test_table_changes_tracking_" + randomNameSuffix();
+
+    private static final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("delta_lake")
+            .setSchema(SCHEMA)
+            .build();
+
+    private File dataDirectory;
+    private HiveMetastore metastore;
+    private AccessTrackingFileSystemFactory fileSystemFactory;
+
+    @Override
+    protected DistributedQueryRunner createQueryRunner()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION).build();
+
+        dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("delta_lake").toFile();
+        metastore = createTestingFileHiveMetastore(dataDirectory);
+        fileSystemFactory = new AccessTrackingFileSystemFactory(new HdfsFileSystemFactory(HDFS_ENVIRONMENT));
+
+        queryRunner.installPlugin(new TestingDeltaLakePlugin(Optional.of(new TestingDeltaLakeMetastoreModule(metastore)), Optional.of(fileSystemFactory), EMPTY_MODULE));
+        queryRunner.createCatalog(
+                "delta_lake",
+                "delta_lake",
+                ImmutableMap.of("delta.enable-non-concurrent-writes", "true"));
+
+        queryRunner.execute("CREATE SCHEMA " + SCHEMA);
+        return queryRunner;
+    }
+
+    @Test
+    public void testTableChangesFileSystemAccess()
+    {
+        String tableName = "table_changes_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (page_url VARCHAR, domain VARCHAR, views INTEGER) WITH (change_data_feed_enabled = true)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url1', 'domain1', 1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url2', 'domain2', 2)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url3', 'domain3', 3)", 1);
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url22' WHERE domain = 'domain2'", 1);
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url33' WHERE views = 3", 1);
+        assertUpdate("DELETE FROM " + tableName + " WHERE page_url = 'url1'", 1);
+        assertQuery("SELECT * FROM " + tableName, "VALUES('url22', 'domain2', 2), ('url33', 'domain3', 3)");
+
+        fileSystemFactory.resetOpenCount();
+        assertQuerySucceeds("SELECT * FROM TABLE(system.table_changes('" + SCHEMA + "', '" + tableName + "'))");
+        assertThat(fileSystemFactory.getOpenCount()).containsAllEntriesOf(
+                ImmutableMap.of(
+                        "00000000000000000000.json", 2,
```

Comment:
if i change 2 -> 1 on this line, the test still passes. i.e. this is a wrong way ot asserting opened file counts